src/main/java/com/elastic/support/diagnostics/commands/RunKibanaQueries.java

```diff
@@ -0,0 +1,309 @@
+        JavaPlatform javaPlatformOs = new JavaPlatform(nodeProfileOs);
+        return javaPlatformOs;
+    }
+
+    /**
+    * this private function will create new object RemoteSystem
+    *
+    * @param  String targetOS
+    * @param  DiagnosticContext context
+    * @return         RemoteSystem object
+    */
+    private RemoteSystem getRemoteSystem(String targetOS, DiagnosticContext context) {
+        RemoteSystem syscmd = new RemoteSystem(
+                            targetOS,
+                            context.diagnosticInputs.remoteUser,
+                            context.diagnosticInputs.remotePassword,
+                            context.diagnosticInputs.host,
+                            context.diagnosticInputs.remotePort,
+                            context.diagnosticInputs.keyfile,
+                            context.diagnosticInputs.pkiKeystorePass,
+                            context.diagnosticInputs.knownHostsFile,
+                            context.diagnosticInputs.trustRemote,
+                            context.diagnosticInputs.isSudo
+                    );
+        ResourceCache.addSystemCommand(Constants.systemCommands, syscmd);
+        return syscmd;
+    }
+
+    /**
+    * On this function we will get create an LocalSystem object
+    * depending the parseOperatingSystem value it will override or not the value of osName with the constant SystemProperties.osName
+    *
+    * @param  String osName
+    * @param  Boolean parseOperatingSystem
+    * @return         LocalSystem object
+    */
+    private LocalSystem getLocalSystem(String osName, Boolean parseOperatingSystem) {
+        if (parseOperatingSystem == true) {
+            osName = SystemUtils.parseOperatingSystemName(SystemProperties.osName);
+        }
+        LocalSystem syscmd = new LocalSystem(osName);
+        ResourceCache.addSystemCommand(Constants.systemCommands, syscmd);
+        return syscmd;
+    }
+
+
+    /**
+    * CheckKibanaVersion (executed before) defined/set the context.elasticRestCalls.
+    * here we will loop and create a List of RestEntry with the queries/APIs that need to be executed
+    * You have two APIs that work differently and and may have or not many pages, so we use the getAllPages function
+    * runQueries called in this function create a json file for each RestEntry set on the 'queries' variable.
+    * Public function so we can test it.
+    *
+    * @param  RestClient client
+    * @param  DiagnosticContext context
+    * @return int
+    */
+    public int runBasicQueries(RestClient client, DiagnosticContext context) {
+
+        int totalRetries = 0;
+        List<RestEntry> queries = new ArrayList<>();
+        
+        for (Map.Entry<String, RestEntry> entry : context.elasticRestCalls.entrySet()) {
+
+            String actionName = entry.getValue().getName().toString();
+            if (actionName.equals("kibana_alerts") || actionName.equals("kibana_detection_engine_find")) {
```

Comment:
Manually listing, in code, the requests that require paging is probably going to be annoying in the long term, but it's fine for now.

In the future, we should probably augment the YAML file that defines these APIs to allow it to specify whether it pages. (No action needed for today)